### PR TITLE
Add TypeKVPairs field type

### PR DESCRIPTION
**Pull request by chrishoffman**
This field type is designed to support key pair type metadata on both the API and the CLI.  Both formats are supported on the API and CLI but the two different formats are optimized for one or the other.  

The API optimized format is map[string]string:

```
...
"foo": {
  "key1": "value1",
  "key2": "value2"
},
...
```

The CLI optimized format is equal sign separated key pairs.

```
 vault write ... foo=key1=value1 foo=key2=value2
```

### CODE REVIEW

File: logical/framework/backend.go [hunk 1]
```diff
@@ -612,6 +612,8 @@ func (t FieldType) Zero() interface{} {
 		return false
 	case TypeMap:
 		return map[string]interface{}{}
+	case TypeKeyPairs:
```
Comment (vishalnayak) [inline logical/framework/backend.go]: Any thoughts on renaming this as `TypeKVPairs`?

**Comment by vishalnayak** [comment]
This looks great to me. The `key` supplied on the CLI can't be base64 encoded, which I think is a decent compromise for the convenience of command line. Also, this obviates the need to use `helper/flag-kv` which is nice!

File: logical/framework/backend.go (hunk 1 above)
Comment (jefferai) [reply logical/framework/backend.go]: +1 from me on TypeKVPairs

**Comment by jefferai** [comment]
@vishalnayak `flag-kv` is used for different things -- it's used with command line flags to generate a map. This is used with command line arguments. Similar goals, but for different parts of the CLI/request.

**Pull request merged**